transport-classes-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java

```diff
@@ -277,13 +278,17 @@ public int registeredChannels() {
     }
 
     private int epollWait(long deadlineNanos) throws IOException {
+        final long result;
         if (deadlineNanos == NONE) {
-            return Native.epollWait(epollFd, events, timerFd, Integer.MAX_VALUE, 0); // disarm timer
+            result = Native.epollWaitWithoutTimer(epollFd, events, timerFd, Integer.MAX_VALUE, 0); // disarm timer
+        } else {
+            long totalDelay = deadlineToDelayNanos(deadlineNanos);
+            int delaySeconds = (int) min(totalDelay / 1000000000L, Integer.MAX_VALUE);
+            int delayNanos = (int) min(totalDelay - delaySeconds * 1000000000L, MAX_SCHEDULED_TIMERFD_NS);
+            result = Native.epollWaitWithoutTimer(epollFd, events, timerFd, delaySeconds, delayNanos);
         }
-        long totalDelay = deadlineToDelayNanos(deadlineNanos);
-        int delaySeconds = (int) min(totalDelay / 1000000000L, Integer.MAX_VALUE);
-        int delayNanos = (int) min(totalDelay - delaySeconds * 1000000000L, MAX_SCHEDULED_TIMERFD_NS);
-        return Native.epollWait(epollFd, events, timerFd, delaySeconds, delayNanos);
+        usedTimer = ((int) result) == 1;
```

Comment:
This will do a signed conversion so won't give the correct value when `result` is negative
```suggestion
        usedTimer = (int) (result & 0xff) == 1;
```